polybrain.el

```diff
@@ -41,92 +41,88 @@ changed."
 (add-hook 'org-brain-visualize-text-hook 'polybrain-visualize-setup)
 
 (defun polybrain--get-point-min-max ()
+  "Get `org-mode' min position and `org-brain-visualize' max position."
   (save-excursion
-    (beginning-of-buffer)
-    (search-forward "--- Entry" nil t)
-    (beginning-of-line)
+    (goto-char org-brain--vis-entry-text-marker)
     (previous-line)
     (let ((max (point)))
       (next-line)
       (next-line)
       (list (point) max))))
 
 (defun polybrain--entry-contents ()
+  "Get text content of the visualized entry."
   (save-excursion
-    (beginning-of-buffer)
-    (search-forward "--- Entry")
+    (goto-char org-brain--vis-entry-text-marker)
     (end-of-line)
-    (forward-char)
-    (let ((beg (point)))
-      (end-of-buffer)
-      (buffer-substring beg (point)))))
+    ;; (forward-char)
+    (buffer-substring (point) (point-max))))
 
 (defun polybrain--save-contents (content)
+  "Replace the entry's text with CONTENT."
   (find-file (org-brain-entry-path org-brain--vis-entry))
-  (beginning-of-buffer)
-  (goto-char
-   (or (save-excursion
-         (while (and (re-search-forward org-brain-keyword-regex nil t) (org-before-first-heading-p))
-           (end-of-line)
-           (forward-char))
-         (point))
-       (point-min)))
-  (insert content)
-  (delete-region (point) (point-max))
-  (save-buffer)
-  (switch-to-buffer (other-buffer (current-buffer) 1)))
+  (seq-let (entry-min entry-max) (org-brain-text-positions org-brain--vis-entry)
```

Comment:
`org-brain-text-positions` gets the min and max positions of the entry text in its file buffer.